src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -1950,6 +1950,10 @@ xocl_kds_xgq_cfg_cus(struct xocl_dev *xdev, xuid_t *xclbin_id, struct xrt_cu_inf
 		 * approach.
 		 */
 		cfg_cu->payload_size = cfg_cu->payload_size * 2;
```

Comment:
We may need to error out if the payload size is more than MAX_CQ_SLOT_SIZE before doubling. Otherwise, we may not be configuring CU properly.
We also need to ignore the streaming arguments if that is causing an issue.